### ISSUE TITLE
Executable does not respect .template-lintrc.js

### ISSUE THREAD

**Issue by john-griffin** [ember-template-lint/ember-template-lint#510]
To get tests passing we defined pending rules in `.template-lintrc.js`. This config is picked up correctly when running the ember tests but when running the `ember-template-lint` executable the config is ignored and it produces errors. 

As the linter executable run is added to `.travis.yml` by default in cli in `3.4` we first noticed this when running on travis and then confirmed by running locally.

**Comment by rwjblue** [comment]
Can you share a snippet of your config? What are the pending module names?

**Comment by john-griffin** [comment]
``` js
'use strict';

module.exports = {
  pending: [
    {
      moduleId: 'app-name/templates/orders/show',
      only: ['no-duplicate-attributes', 'quotes', 'attribute-indentation']
    }
  ],
  extends: 'recommended'
};
```

Here is a stripped down version, `app-name` is replace by our actual app name.

**Comment by rwjblue** [comment]
Yeah, the issue here is that the module name during the build (as part of ember-cli-template-lint) is not the actual on disk location. When running the included binary you see the **real** path (`app/templates/orders/show`), whereas in your build and config you have `app-name`.

**Comment by john-griffin** [comment]
Ah thanks for explaining that. `app` works for the executable and `app-name` works in the tests. Is there a way to get linting passing in both of these contexts with the same config?

**Comment by rwjblue** [comment]
Yeah, this is definitely not ideal. Sorry about that. 

The simplest thing to do would be to have both entries in your pending list 😢.

**Comment by john-griffin** [comment]
Ok no worries makes sense. Trying a few things out and looking at the source it appears that minimatch works if you don't care about the specific rules, eg:

``` js
'use strict';
const Minimatch = require('minimatch').Minimatch;

module.exports = {
  pending: [
    new Minimatch("*/templates/orders/show")
  ],
  extends: 'recommended'
};
```

Would you be interested in a PR that adds minimatch support at the `moduleId` level? So it could look something like this:

``` js
'use strict';
const Minimatch = require('minimatch').Minimatch;

module.exports = {
  pending: [
    {
      moduleId: new Minimatch("*/templates/orders/show"),
      only: ['no-duplicate-attributes', 'quotes', 'attribute-indentation']
    }
  ],
  extends: 'recommended'
};
```

Or do you have a different suggestion on how to move this forward?

**Comment by jacobq** [comment]
In case it's useful to others here's a quick hack to implement the dual entry solution mentioned above:

`.template-lintrc.js`
```js
'use strict';

const pending = [
  // (paste output here)
  {
    moduleId: "my-ember-app/foo",
    // ...
  }, 
  // ...
];

module.exports = {
  extends: 'recommended',
  // ...
  pending: pending.concat(pending.map(item => {
    item.moduleId = item.moduleId.replace('my-ember-app', 'app');
    return item;
  }))
};
```

**Comment by panthony** [comment]
I'm kind of lost with this template lint.

- I have indentation issues on a template -> ok
- I create the pending config using 'print-failures' that I had a hard time to find -> ok
- I'm doing the workaround 'my-ember-app'/'app' here to workaround an issue -> ok

But now... running `ember test` it complains like this:

>  -:-  error  Pending module (`my-ember-app/components/my-component/template`) passes all rules. Please remove `my-ember-app/components/template` from pending list.  undefined

If I remove the pending rule then `ember test` will complain with the linter issues.

Any ideas?

**Comment by panthony** [comment]
I figured this is with the `1.0.0` release because of:

> https://github.com/ember-template-lint/ember-template-lint/pull/599/files

Problem:

>  ember template-lint:print-failing

Now generate this for me:

```
  {
    "moduleId": "my-app/components/my-component/template",
    "only": [
      null
    ]
  }
```

:sob:
